matchzoo/contrib/models/hbmp.py

```diff
@@ -0,0 +1,121 @@
+        concat1 = keras.layers.Subtract()([embed_sen_left, embed_sen_right])
+        concat1_abs = keras.layers.Lambda(lambda x: abs(x))(concat1)
+        concat2 = keras.layers.Multiply()([embed_sen_left, embed_sen_right])
+        concat = keras.layers.Concatenate(axis=1)(
+            [embed_sen_left, embed_sen_right, concat1_abs, concat2])
+
+        # multiply perception layers to classify
+        mlp_out = self._classifier(concat, mlp_num_layers=self._params['mlp_num_layers'], mlp_num_units=self._params[
+            'mlp_num_units'], drop_rate=self._params['dropout_rate'], leaky_relu_alpah=self._params['alpha'])
+        out = self._make_output_layer()(mlp_out)
+
+        self._backend = keras.Model(
+            inputs=[input_left, input_right], outputs=out)
+
+    def _classifier(
+        self,
+        input_: typing.Any,
+        mlp_num_layers: int,
+        mlp_num_units: list,
+        drop_rate: float,
+        leaky_relu_alpah: float
+    )-> typing.Any:
+        for i in range(mlp_num_layers - 1):
+            input_ = keras.layers.Dropout(rate=drop_rate)(input_)
+            input_ = keras.layers.Dense(mlp_num_units[i])(input_)
+            input_ = keras.layers.LeakyReLU(alpha=leaky_relu_alpah)(input_)
+
+        return input_
+
+    def _sentence_encoder(
+        self,
+        input_: typing.Any,
+        lstm_num_units: int,
+        drop_rate: float
+    ) -> typing.Any:
+        """
+        stack three BiLSTM MaxPooling blocks as a hierarchical structure.
+        concatenate the output of three blocs as the input sentence embedding.
+        each BiLSTM layer reads the input sentence as the input.
+        each BiLSTM layer except the first one is initialized(the initial hidden state and the cell state) with the final state of the previous layer.   
+        """
+        layer1 = keras.layers.Bidirectional(keras.layers.LSTM(units=lstm_num_units, return_sequences=True,
+                                                              return_state=True, dropout=drop_rate, recurrent_dropout=drop_rate), merge_mode='concat')(input_)
+        layer1_maxpooling = keras.layers.GlobalMaxPooling1D()(layer1[0])
+
+        layer2 = keras.layers.Bidirectional(keras.layers.LSTM(units=lstm_num_units, return_sequences=True, return_state=True,
+                                                              dropout=drop_rate, recurrent_dropout=drop_rate), merge_mode='concat')(input_, initial_state=layer1[1:5])
+        layer2_maxpooling = keras.layers.GlobalMaxPooling1D()(layer2[0])
+
+        layer3 = keras.layers.Bidirectional(keras.layers.LSTM(units=lstm_num_units, return_sequences=True, return_state=True,
+                                                              dropout=drop_rate, recurrent_dropout=drop_rate), merge_mode='concat')(input_, initial_state=layer2[1:5])
+        layer3_maxpooling = keras.layers.GlobalMaxPooling1D()(layer3[0])
```

Comment:
1. variable naming issues.
2. line formatting (see (here)[https://www.python.org/dev/peps/pep-0008/#maximum-line-length])

Reply:
ok, I have corrected these issues, please review it.